### PR TITLE
stylo: Properly serialize grid shorthand

### PR DESCRIPTION
**Pull request by canova**
It was failing to serialize correctly when grid was `<'grid-template'>` or some different edge cases.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17630)

### CODE REVIEW

File: components/style/properties/shorthand/position.mako.rs [hunk 1]
```diff
@@ -495,16 +495,30 @@
         })
     }
 
+    /// Returns true if every sub property value of `grid` shorthand is initial.
+    impl<'a> LonghandsToSerialize<'a> {
+        fn is_initial(&self) -> bool {
```
Comment (canaltinova) [inline components/style/properties/shorthand/position.mako.rs]: I'm not happy about this method but this seems the best way to check it. Without that we can't serialize this shorthand correctly. 😕
